docs/architecture/psa-migration/md-cipher-dispatch.md

```diff
@@ -73,6 +73,7 @@ The following configuration options are described as experimental, and are likel
 
 * `MBEDTLS_PSA_CRYPTO_CLIENT`: “This interface is experimental and may change or be removed without notice.” In practice we don't want to remove this, but we may constrain how it's used.
 * `MBEDTLS_PSA_CRYPTO_DRIVERS`: “This interface is experimental. We intend to maintain backward compatibility with application code that relies on drivers, but the driver interfaces may change without notice.” In practice, this may mean constraints not only on how to write drivers, but also on how to integrate drivers into code that is platform code more than application code.
+* `MBEDTLS_PSA_CRYPTO_CONFIG`: “This feature is still experimental and is not ready for production since it is not completed.” We may want to change this, for example, to automatically enable more mechanisms (although this wouldn't be considered a backward compatibility break anyway, since we don't promise that you will not get a feature if you don't enable its `PSA_WANT_xxx`).
```

Comment:
I think you and me know what this is about, but for the benefit of other readers it might be worth being more explicit: we may want to make it so that all hashes (for example) that are available in legacy form are also available in PSA.

(Same for the previous bullet point btw: in particular, we might want to require that hash accelerators don't need an explicit init function.)

Reply:
I don't see why this should be mentioned here. It's up to the later analysis to figure out what we do want to change.

Reply:
> It's up to the later analysis to figure out what we do want to change.

Of course, and that's why I used the same wording ("We may want to") as the current text. I just think "automatically enable more mechanisms" may not be clear enough and we could make that part more explicit: "automatically enable in PSA all hashes that are available on the legacy side".

(Same for the previous sentence, "[constraints] on how to integrate drivers into code that is platform code more than application code" sounds rather abstract to me, and I would be unlikely to figure out it may include things like require that some accelerators don't need init to be called, if we hadn't discussed this beforehand.)

Reply:
I'd rather not make those changes because they would distract from the analysis. The point is to start from the requirements and derive the conclusions. I want to avoid injecting presupposed conclusions into the initial analysis of individual requirements.

Reply:
Ok, I see your point.